kong/clustering/wrpc_data_plane.lua

```diff
@@ -245,31 +252,44 @@ local function communicate_impl(dp)
   end
 end
 
-local communicate_loop
 
-function communicate(dp, reconnection_delay)
-  dp.log_suffix = " [" .. dp.conf.cluster_control_plane .. "]"
-  return ngx.timer.at(reconnection_delay or 0, communicate_loop, dp)
-end
+local function communicate(dp, reconnection_delay)
 
-function communicate_loop(premature, dp)
-  if premature then
-    -- worker wants to exit
-    return
-  end
+  local communicate_loop = function(premature, dp)
```

Comment:
why not just
```suggestion
  local  function communicate_look(premature, dp)
```

Reply:
I prefer not to make it nested. That's the reason I wrote "local communicate_loop" and defined it later.

Reply:
I agree. But I also don't like forward declare vars, which makes the code too hard to read.